filecoin-proofs/src/commitment_reader.rs

```diff
@@ -7,11 +7,13 @@ use rayon::prelude::{ParallelIterator, ParallelSlice};
 
 use crate::{constants::DefaultPieceHasher, pieces::piece_hash};
 
+const BUFFER_SIZE: usize = 4096;
```

Comment:
Have you tried bumping it even further, for example, up to 1MiB-8MiB?
Even with the 4KiB, it is just 64 SHA invocations, the branch predictor is probably just warming up at that point.

Reply:
I've tried 8MiB it didn't make a difference.

I think yet another difference would make it if the hashing would be done in parallel. But a simple "par chunker" didn't really help, as then the pieces are to small. One need to implement it manually. I decided that it's not worth it and that this improvement is already good enough for how simple it is.